torch/_dynamo/config.py

```diff
@@ -69,6 +69,14 @@
 # legacy config, does nothing now!
 dynamic_shapes = True
 
+# Experiment to measure the regression from setting the first dimension
+# as dynamic. The benefit is that our graphs become more cacheable,
+# potentially allowing us to recover the regression and even achieve
+# a net improvement in aggregate. THIS IS NOT LANDABLE. We will likely
+# default this to False and slowly roll out with a JK. I'm setting this
+# to True for now so I can run the benchmark suites more easily.
+dynamic_first_dimension = True
```

Comment:
You should explicitly say how the precedence of this compares to assume_static_by_default. Also you need to say what this does when False (in particular, if `assume_static_by_default = False`, marking this False doesn't cause the first dim to be static.) A rename could help, something like `assume_outer_dim_static_by_default = False`